### DISCUSSION TITLE
How to pass document attributes at build time

### DISCUSSION THREAD

**Discussion by brucellino** [docToolchain/docToolchain#1410] [Q&A]
Hello! I have a few Asciidoc document attributes which I would like to pass at build time, instead of writing them in the configuration file.

I can see that there is an attributes block set in the the microsite task - will this work for the PDF? I can't see where to write it though - there's no `pdf.with {}`.

Thanks!

**Comment by rdmueller** [comment]
Good question.
The easiest way is to generate a `_config.adoc` file with all needed attributes and include it.
since `doctoolchainConfig.groovy` is a groovy script, you could use it to read environment variables or gradle parameters and generate the confige file here. This would be quite flexible.

Another place where this could be handled is https://github.com/docToolchain/docToolchain/blob/ng/scripts/AsciiDocBasics.gradle#L155
But there isn't code which sets them dynamically.
We could add a block like this which adds some attributes from the `doctoolchainConfig.groovy`:
https://github.com/docToolchain/docToolchain/blob/ng/scripts/AsciiDocBasics.gradle#L310C1-L315C1

**Reply by ponchofiesta** [reply]
Maybe it would be an option to support .asciidoctorconfig file as introduced by the AsciiDoc IntelliJ plugin: https://intellij-asciidoc-plugin.ahus1.de/docs/users-guide/features/advanced/asciidoctorconfig-file.html

**Reply by ponchofiesta** [reply]
I'd like a way to pass attributes via CLI parameters like `doctoolchain . publishToConfluence -a someVar=blubb`.

**Reply by rdmueller** [reply]
I always thought that the .asciidoctorconfig file is a good solution for IDEs and previews. Now that I think about it, it could also make sense for docToolchain. Who wants to implement it?

**Reply by rdmueller** [reply]
regarding cli parameters. Yes, we should implement this. Who wants to do it?

**Reply by ponchofiesta** [reply]
One problem with .asciidoctorconfig might be it seems to be some special format of the IntelliJ plugin. For example an `ifdef` condition didn't work. So the possibilities are limited, I think.

Unfortunately wrong language. I don't know nothing about Groovy and Gradle.
